api_tests/nodes/views/test_node_alternative_citations.py

```diff
@@ -0,0 +1,1098 @@
+from tests.base import ApiTestCase
+
+from tests.factories import (
+    ProjectFactory,
+    RegistrationFactory,
+    AuthUserFactory,
+    AlternativeCitationFactory
+)
+
+class TestCreateAlternativeCitations(ApiTestCase):
+    def setUp(self):
+        super(TestCreateAlternativeCitations, self).setUp()
+        self.user = AuthUserFactory()
+        self.user_two = AuthUserFactory()
+        self.user_three = AuthUserFactory()
+
+        self.public = ProjectFactory(creator=self.user, is_public=True)
+        self.private = ProjectFactory(creator=self.user, is_public=False)
+        self.citation = AlternativeCitationFactory(name="name", text="text")
+        self.public.alternativeCitations.append(self.citation)
+        self.private.alternativeCitations.append(self.citation)
+        self.public.add_contributor(self.user_two, permissions=[permissions.READ, permissions.WRITE], visible=True, save=True)
+        self.private.add_contributor(self.user_two, permissions=[permissions.READ, permissions.WRITE], visible=True, save=True)
+        self.public.save()
+        self.private.save()
+        self.public_url = '/{}nodes/{}/citations/'.format(API_BASE, self.public._id)
+        self.private_url = '/{}nodes/{}/citations/'.format(API_BASE, self.private._id)
+        self.citation.public_url = self.public_url + '{}/'.format(self.citation._id)
+        self.citation.private_url = self.private_url + '{}/'.format(self.citation._id)
+
+    def payload(self, name=None, text=None, id=None):
+        payload = {'data': {
+            'type': 'citations',
+            'attributes': {
+                'name': name,
+                'text': text
+                }
+            }
+        }
+        if id is not None:
+            payload['data']['id'] = id
+        return payload
+
+    def test_add_citation_admin_public(self):
+        res = self.app.post_json_api(self.public_url, self.payload(name='Test', text='Citation'), auth=self.user.auth)
+        assert_equal(res.status_code, 201)
+        self.public.reload()
+        assert_equal(len(self.public.alternativeCitations), 2)
+
+    def test_add_citation_admin_private(self):
+        res = self.app.post_json_api(self.private_url, self.payload(name='Test', text='Citation'), auth=self.user.auth)
+        assert_equal(res.status_code, 201)
+        self.private.reload()
+        assert_equal(len(self.private.alternativeCitations), 2)
+
+    def test_add_citation_non_admin_public(self):
+        res = self.app.post_json_api(self.public_url, self.payload(name='Test', text='Citation'), auth=self.user_two.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+        self.public.reload()
+        assert_equal(len(self.public.alternativeCitations), 1)
+
+    def test_add_citation_non_admin_private(self):
+        res = self.app.post_json_api(self.private_url, self.payload(name='Test', text='Citation'), auth=self.user_two.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
```

Comment:
For every test that returns a 400-type error, please also assert that you're getting the correct error message so that you know the test is failing for the right reason.